.github/workflows/ci.yml

```diff
@@ -124,6 +124,11 @@ jobs:
       if: matrix.platform.cross == false
       run: cargo build --locked --workspace --all-targets
 
+    - name: Check dependent crate
+      if: matrix.platform.cross == false # TODO: unsure if this would matter
+      run: cargo check
+      working-directory: test_dependent_crate
+
```

Comment:
This becomes unnecessary if we delete Cargo.lock per guidelines for crates.

Reply:
Deleting the Cargo.lock updates the async-trait, where it would seem that upgrade from 0.1.42 to 0.1.43 surfaces this error. This can be tested also by specifying the version requirement for `async-trait` as `=0.1.43` with minimal changes to Cargo.lock (async-trait, syn).